sound/pci/hda/hda_controller.c

```diff
@@ -592,7 +592,7 @@ static int azx_pcm_open(struct snd_pcm_substream *substream)
 		err = -EBUSY;
 		goto unlock;
 	}
-	runtime->private_data = azx_dev;
+	substream->private_data = azx_dev;
```

Comment:
You need a macro here to avoid hard-coding where the stream is stored.
You have a get, you need a set

Reply:
Was thinking about it.

Agree, let's do it.